.github/workflows/conformance-suites.yml

```diff
@@ -22,7 +22,7 @@ jobs:
       - uses: actions/checkout@v4.1.1
         with:
           persist-credentials: false
-          ref: ${{ github.event.pull_request.head.sha }}
+          ref: ${{ github.event.pull_request.merge_commit_sha }}
```

Comment:
Info [here](https://docs.github.com/en/free-pro-team@latest/rest/pulls/pulls?apiVersion=2022-11-28#get-a-pull-request) leads me to believe `merge_commit_sha` may be null until a test merge commit is successfully generated, which I assume would cause this checkout action to fail. 

In practice, there's enough of a delay between us pushing an update to a PR and approving the runs that we probably won't encounter incomplete mergeability checks, but I'll add a step to fail if `mergeable` is not `true`. If it fails, I believe we can just rerun that action after a short period once the mergeability checks complete and it should continue as normal.